File: src/AoC16/s16e01-python/s16e01/part2.py
"""Part 2 solution runner."""

import sys

from s16e01 import solve_part2


def main():
    """Read input from stdin and print part 2 result."""
    input_text = sys.stdin.read()
    result = solve_part2(input_text)
    print(result)


if __name__ == "__main__":
    main()

File: src/AoC16/s16e01-julia/src/part2.jl
#!/usr/bin/env julia

# Get the directory containing this script
const SCRIPT_DIR = @__DIR__

# Include the library module
include(joinpath(SCRIPT_DIR, "lib.jl"))
using .S16E01

function main()
    # Read input from stdin
    input = read(stdin, String)
    result = solve_part2(input)
    println(result)
end

# Run main if this script is executed directly
if abspath(PROGRAM_FILE) == @__FILE__
    main()
end
